GoogleForCreators/web-stories-wp
# New Story Details Modal: Upload/Edit Featured Media From Cover Preview

## Summary

Adds the functionality to upload and edit the cover preview for the story. This should work exactly the same as the button in the document pane.

## Relevant Technical Choices

There is a [shared component used for uploading media](https://github.com/GoogleForCreators/web-stories-wp/blob/928311c162f7d7f0454364755e689ef099badb33/packages/design-system/src/components/mediaInput/index.js#L159-L260) that is used in the document panel.

This component does a lot (renders the story preview and the media upload button). But in an attempt to not implement the same functionality, I added a new variant which removes the story image preview.

Out of scope for this PR, but it may be beneficial to separate the story preview and the upload media button into two separate components.

## User-facing changes

|Before|After|
|--|--|
|<img width="325" alt="image" src="https://user-images.githubusercontent.com/22185279/155811046-68aea4b9-54a5-4626-bd89-00957836f198.png">|![upload-media](https://user-images.githubusercontent.com/22185279/155811500-915d0f7c-b2c2-4486-95c9-27c53ca4f6bd.gif)|

## Testing Instructions






This PR can be tested by following these steps:

1. Enable the new publish modal experiment
2. Open the editor
3. Click the `Publish` button. Should see the new button visible on the story preview
4. Click the button. Should open a modal to select or upload media.
5. Should be able to select, upload, and crop media.
6. Selecting new media should upload the story preview.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #10584
